var/spack/repos/builtin/packages/hpcc/package.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+import re
+import platform
+
+
+class Hpcc(MakefilePackage):
+    """HPC Challenge is a benchmark suite that measures a range memory access
+    patterns.
+    The HPC Challenge benchmark consists of basically 7 tests:
+        1) HPL - the Linpack TPP benchmark which measures the floating point
+           rate of execution for solving a linear system of equations.
+        2) DGEMM - measures the floating point rate of execution of double
+           precision real matrix-matrix multiplication.
+        3) STREAM - a simple synthetic benchmark program that measures
+           sustainable memory bandwidth (in GB/s) and
+           the corresponding computation rate for simple vector kernel.
+        4) PTRANS (parallel matrix transpose) - exercises the communications
+           where pairs of processors communicate
+           with each other simultaneously. It is a useful test of the total
+           communications capacity of the network.
+        5) RandomAccess - measures the rate of integer random updates of memory
+           (GUPS).
+        6) FFT - measures the floating point rate of execution of double
+           precision complex one-dimensional Discrete Fourier Transform (DFT).
+        7) Communication bandwidth and latency - a set of tests to measure
+           latency and bandwidth of  a number of simultaneous communication
+           patterns; based on b_eff (effective bandwidth benchmark)."""
+
+    homepage = "http://icl.cs.utk.edu/hpcc"
+    url = "http://icl.cs.utk.edu/projectsfiles/hpcc/download/hpcc-1.5.0.tar.gz"
+    hg = "http://hg.code.sf.net/p/hpcc/code"
+
+    version('develop', hg=hg)
+    version('1.5.0', sha256='0a6fef7ab9f3347e549fed65ebb98234'
+                            'feea9ee18aea0c8f59baefbe3cf7ffb8', preferred=True)
+
+    variant(
+        'fft', default='internal', description='FFT library to use',
+        values=('internal', 'fftw2', 'mkl'), multi=False
```

Comment:
We should never use internal. I think you should remove this variant and add a dependency on `fftw-api`, which can be either `fftw` or `intel-mkl`.

Reply:
Internal here is not like charm++ within namd source code, It is fairly integrated within the code. The internal FFT is derived from FFTE and I don't feel comfy to simply hide it but at the same time I don't want to pull it out at this point.

Reply:
That's fine then, as long as it isn't a vendored copy.

Reply:
I take a look into FFTE library it does not have any building system, so I would guess that intended use is to integrate it into the the code.